core/src/epicli/data/common/ansible/playbooks/roles/preflight/tasks/main.yml

```diff
@@ -11,6 +11,56 @@
     fail_msg: "Your architecture {{ ansible_architecture }} is not supported at {{ ansible_distribution }} by following components in inventory -> {{ unsupported_components_checklist }}."
     success_msg: "Your architecture {{ ansible_architecture }} is supported at {{ ansible_distribution }} by all components in inventory."
 
+- name: Check if enabled applications are supported by the architecture
+  block:
+    - name: Include application vars from applications role
+      include_vars:
+        file: roles/applications/vars/main.yml
+        name: k8s_apps
+
+    - name: Check if enabled applications are supported by the architecture
+      assert:
+        that: _unsupported_enabled_apps | length == 0
+        fail_msg: "Following application(s) are not supported by {{ _k8s_arch }} and cannot be installed: {{ _unsupported_enabled_apps | map(attribute='name') | join(', ') }}"
+        success_msg: "All enabled applications are supported by {{ _k8s_arch }}"
+      vars:
+        _k8s_arch: >-
+          {{ hostvars[groups['kubernetes_master'][0]]['ansible_architecture'] }}
+        _defined: >-
+          {{ k8s_apps.specification.applications | selectattr('enabled', 'defined') | list }}
+        _undefined: >-
+          {{ k8s_apps.specification.applications | selectattr('enabled', 'undefined') | list }}
+        _enabled_apps: >-
+          {{ (_defined | selectattr('enabled') | list) + _undefined }}
+        _unsupported_enabled_apps: >-
+          {{ _enabled_apps | selectattr('name', 'in', unsupported_apps[_k8s_arch]) | list }}
+  run_once: true
+  delegate_to: localhost
+  when:
+    - not is_upgrade_run # only for apply
+    - groups.kubernetes_master is defined
+    - groups.kubernetes_master | length > 0
+
+- name: Check if PostgreSQL extentions and architecture are not compatible
```

Comment:
How about doing this the same way as for `unsupported_apps`, i.e. have a list for PostgreSQL extensions like:
```
unsupported_postgresql_extensions:
  x86_64: []
  aarch64:
    - pgbouncer
    - replication
```

Reply:
It seems not to be necessary as applications and components may be extended, but not sure that it's so for PostgreSQL extensions.

Reply:
Each extension requires different steps and there is not only OS pacages installation, so for such approach I would move tasks per extension to separate files, but it will require refactoring and may introduce new bugs. Let's leave it as it is at least until this role is refactored for upgrades.